gm4_vecto_shamir/data/gm4_vecto_shamir/functions/vehicle/spawn_boat.mcfunction

```diff
@@ -0,0 +1,15 @@
+#run from vehicle/check_boat
+#@s = player with shamir equipped, sneaking above water
+
+tag @s add gm4_ml_invehicle
+
+# spawn boat
+summon minecraft:boat ~ ~ ~ {Invulnerable:1b,Tags:["gm4_ml_vehicle"]}
+execute rotated as @p[tag=gm4_ml_invehicle] run tp @e[tag=gm4_ml_vehicle] ~ ~ ~ ~ ~
+
+# link player to boat with id
+execute store result score @s gm4_ml_vecto_id run data get entity @s UUID[0]
```

Comment:
players should get their `gm4_ml_vecto_id` score via a join advancement, so it only ever happens once per player